### DISCUSSION TITLE
How to block light

### DISCUSSION THREAD

**Discussion by CyberFranck** [StereoKit/StereoKit#461] [Q&A] [answered]
The light from the scene is passing trough the objects killing the immersion of the model. Is there and option to make the light works properly like ambient occlusion or something ?

Specially i need to add custom lights source somehow inside the model.

**Comment by maluoi** [comment, marked as answer]
StereoKit doesn't do complex lighting terribly well, instead if focuses on performance for mobile headsets! Dynamic shadow casting and lighting is generally a pretty expensive feature set that will bring a standalone headset to its knees, so StereoKit doesn't have this built-in.

My recommendations would be to bake your lighting in advance with some kind of lightmapper, or try and get creative with your lighting effects. Blob shadows and that sort of thing.

If realism and complex lighting are truly essential to your project, and you are unbound from mobile XR headsets, there's a good chance you might want the more fully featured rendering pipeline of a game engine.

**Comment by CyberFranck** [comment]
I'll have to go with something else. My choice was between StereoKit which is a very simple API and Unity which is a much more complex tool that doesn't work out of the box without lots and lots of code.
